### PR TITLE
Add test for expansion_panel_list.expansion_panel_list_radio.0_test.dart

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by TahaTesser** [comment]
It seems like there is a conflict

File: examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    expectExpandedItem(tester, 2); // The default expanded item is 2.
+
+    // Open all the panels one by one.
+    for (int i = 0; i < 8; i++) {
+      await tester.ensureVisible(find.byType(ExpandIcon).at(i));
+      await tester.tap(find.byType(ExpandIcon).at(i));
+      await tester.pumpAndSettle();
+      expectExpandedItem(tester, i);
+    }
```
Comment (TahaTesser) [inline examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart:39]: `expectExpandedItem` loops through 8 items and then there is nested loop? 

Please put "expect" in the test itself and put any loops in the test for readability.

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart (hunk 1 above)
Comment (ValentinVignal) [reply examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart]: Sure, I changed it in [refactor: Don't use utility method](https://github.com/flutter/flutter/pull/151730/commits/dc5c593edf3d851782e7cf1a49a7b4f08d7fd0ff)

File: examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('ExpansionPanelList.radio can expand on item at the time', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart:10]:
```suggestion
  testWidgets('ExpansionPanelList.radio can expand on item at the time', (WidgetTester tester) async {
```

Is this a correct tile? "expand on item at the item"?

Comment (ValentinVignal) [reply examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart:10]: Sorry, there is a typo, *one item at the time.

Good catch

Comment (ValentinVignal) [reply examples/api/test/material/expansion_panel/expansion_panel_list.expansion_panel_list_radio.0_test.dart]: This is fixed in [Fix typo in test description](https://github.com/flutter/flutter/pull/151730/commits/efb8f865ea7a07e985303a8d771b105d3b4883de)

**Review by TahaTesser — APPROVED** [review]
LGTM

**Review by bleroux — APPROVED** [review]
LGTM!

**Pull request merged**